python/hail/methods/statgen.py

```diff
@@ -5,12 +5,102 @@
 from hail.linalg import BlockMatrix
 from hail.typecheck import *
 from hail.utils import wrap_to_list, new_temp_file, info
-from hail.utils.java import handle_py4j
+from hail.utils.java import handle_py4j, joption
 from .misc import require_biallelic
 from hail.expr import functions
 import hail.expr.aggregators as agg
 
 
+@handle_py4j
+@typecheck(dataset=MatrixTable,
+           maf=nullable(expr_numeric),
```

Comment:
This needs to be `nullable(Float64Expression)`.

Reply:
Do we not want to support other types of numeric (convertible) expressions? the expr_numeric typechecker is nice because it allows plain python ints and floats etc., but if that doesn't make sense here then I can change. (at the very least, should I have Float32Expression and Float64Expression?)

Reply:
The issue is on the Java side for the IBD code. It coerces it to a double on line 349 of IBD.scala. We need to figure out how to do numeric type promotion in Python. But for now, I wanted you to have what will work now.